Assets/Python/OOSLogger.py

```
@@ -1,190 +1,462 @@
+        szPlayerName = unicode(szPlayerName, 'latin-1', 'replace')
+    # Filesystem-safe filename (ASCII-only) and Unicode path components
+    safeName = u''.join(
+        c if (ord(c) < 128 and (c.isalnum() or c in (u' ', u'-', u'_', u'.')))
+        else u'_'
+        for c in szPlayerName
+    )
+    if len(safeName) > 60:
+        safeName = safeName[:60].rstrip()
+    # Build Unicode log dir using filesystem encoding
+    fsenc = (sys.getfilesystemencoding()
+             or locale.getpreferredencoding()
```

Ruff (0.12.2)

Undefined name `locale`

(F821)